src/libstd/io/mod.rs

```diff
@@ -1899,6 +1906,12 @@ impl<T: Read> Read for Take<T> {
     unsafe fn initializer(&self) -> Initializer {
         self.inner.initializer()
     }
+
+    fn read_to_end(&mut self, buf: &mut Vec<u8>) -> Result<usize> {
+        let reservation_size = cmp::min(self.limit, 32) as usize;
```

Comment:
`32` is very low, why did you choice to only read 32 bytes each time ?

Reply:
This was not my choice - I only preserved the original value.